go/lint/action.yml

```diff
@@ -0,0 +1,126 @@
+name: "Go Lint"
+description: "Lint Go Repo"
+
+inputs:
+  FLIPPCIRCLECIPULLER_REPO_TOKEN:
+    description: Flipp circleci repo token
+    required: true
+  BUF_BUILD_USER:
+    description: Buf CI user stored as secret
+    required: false
+  BUF_BUILD_API_TOKEN:
+    description: Buf API token stored as secret
+    required: false
+  SLACK_CHANNEL_ID:
+    description: The Slack channel ID(s) to send the data to
+    required: false
+  SLACK_BOT_TOKEN:
+    description: The Slack bot token to pass the data to
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set branch variable
+      id: set-branch
+      shell: bash
+      run: |
+        if [ "${{ github.event_name }}" == "pull_request" ]; then
+          branch="${{ github.event.pull_request.head.ref }}"
+        else
+          branch="${{ github.ref }}"
+          branch="${branch#refs/heads/}"
+        fi
+        echo "branch=${branch}" >> $GITHUB_OUTPUT
+    - name: Get short sha
+      id: get-short-sha
+      run: echo "short_sha=`echo ${GITHUB_SHA::7}`" > $GITHUB_OUTPUT
+      shell: bash
+    - name: Configure Go environment
+      uses: wishabi/github-actions/go/configure@v0
+      with:
+        FLIPPCIRCLECIPULLER_REPO_TOKEN: ${{ inputs.FLIPPCIRCLECIPULLER_REPO_TOKEN }}
+        BUF_BUILD_USER: ${{ inputs.BUF_BUILD_USER }}
+        BUF_BUILD_API_TOKEN: ${{ inputs.BUF_BUILD_API_TOKEN }}
+        FETCH_DEPTH: 0
+    - name: Generate local protos
+      run: make generate-buf
+      shell: bash
```

Comment:
we lose (a little bit/ a lot of) time regenerating protos over and over for each action.
Maybe we can think about cutting that down in a future iteration of these actions

Reply:
For another day, but what about having CI generate these (so we ensure accuracy) and commit them? Then the repo is in a usable state when pulled, but we don't have to worry about drift between our code and generated artifacts.

Reply:
That could be another option. Maybe we should bring this back to the wider team to discuss.

Reply:
Agreed. Are we good to leave this out of scope for now and iterate on this after?